### PR TITLE
Switch to pydata theme for docs

### LINKED ISSUE #1150: Solution to building docs for old versions
### Summary

Building docs requires importing all the modules in mantidimaging. To get the docs for old releases sphinx-multiversion checks out each release, and builds those docs. So to build the docs we need an environment with that can import all the old versions of mantidimaging.

Its only a matter of time until it becomes impossible to have a single python environment that can import all versions of mantidimaging. Currently we are keeping sarepy and hence the dtasev channel in our environment to allow the old docs to build (  #1149 ).

### Steps To Reproduce

Remove sarepy from the environment and run `python3 ./setup.py docs` 

### Expected Behaviour
Builds versioned docs

### Current Behaviour
```
ModuleNotFoundError: No module named 'sarepy'
```

### Context

Happens when we update the environment.

### Failure Logs

```
Exception occurred:
  File "/tmp/tmpctpwl4hm/c1aa0b13a9d3b8c7fba3cecaedb638dfb05d3b49/mantidimaging/core/operations/remove_all_stripe/remove_all_stripe.py", line 8, in <module>
    from sarepy.prep.stripe_removal_original import remove_all_stripe
ModuleNotFoundError: No module named 'sarepy'
The full traceback has been saved in /tmp/sphinx-err-u0b78ezd.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
Traceback (most recent call last):
Running setup.py internal_docs_api
  File "/usr/share/miniconda3/envs/mantidimaging-dev/bin/sphinx-multiversion", line 8, in <module>
    sys.exit(main())
Running setup.py internal_docs
Running sphinx-multiversion
  File "/usr/share/miniconda3/envs/mantidimaging-dev/lib/python3.8/site-packages/sphinx_multiversion/main.py", line 363, in main
    subprocess.check_call(cmd, cwd=current_cwd, env=env)
  File "/usr/share/miniconda3/envs/mantidimaging-dev/lib/python3.8/subprocess.py", line 364, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '('/usr/share/miniconda3/envs/mantidimaging-dev/bin/python', '-R', '-m', 'sphinx', '-D', 'smv_metadata_path=/tmp/tmpctpwl4hm/versions.json', '-D', 'smv_current_version=2.2.0', '-c', '/home/runner/work/mantidimaging/mantidimaging/docs', '/tmp/tmpctpwl4hm/c1aa0b13a9d3b8c7fba3cecaedb638dfb05d3b49/docs', '/home/runner/work/mantidimaging/mantidimaging/docs/build/html/2.2.0')' returned non-zero exit status 2.
Traceback (most recent call last):
  File "setup.py", line 216, in <module>
    setup(
  File "/usr/share/miniconda3/envs/mantidimaging-dev/lib/python3.8/site-packages/setuptools/__init__.py", line 153, in setup
    return distutils.core.setup(**attrs)
  File "/usr/share/miniconda3/envs/mantidimaging-dev/lib/python3.8/distutils/core.py", line 148, in setup
    dist.run_commands()
  File "/usr/share/miniconda3/envs/mantidimaging-dev/lib/python3.8/distutils/dist.py", line 966, in run_commands
    self.run_command(cmd)
  File "/usr/share/miniconda3/envs/mantidimaging-dev/lib/python3.8/distutils/dist.py", line 985, in run_command
    cmd_obj.run()
  File "setup.py", line 115, in run
    subprocess.check_call(command_sphinx_multiversion)
  File "/usr/share/miniconda3/envs/mantidimaging-dev/lib/python3.8/subprocess.py", line 364, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['sphinx-multiversion', 'docs', 'docs/build/html']' returned non-zero exit status 1.
Error: Process completed with exit code 1.
```

### LINKED ISSUE #2080: Switch sphinx theme to PyData
### Desired Behaviour
https://pydata-sphinx-theme.readthedocs.io/
This the theme used by numpy and similar

It does version switching as part of the theme
https://pydata-sphinx-theme.readthedocs.io/en/stable/user_guide/version-dropdown.html
So we could drop `sphinx_multiversion`

We change how we push to the gh-pages branch in so that we can retain old versions
see .github/actions/publish-docs/action.yml

### PR DESCRIPTION
**Pull request by samtygier-stfc**
### Issue
Closes #2080
Also closes #1150

### Description

Add pydata sphinx package, and set as theme
Remove sphinx_multiversion
Move doc publishing from setup.py into action
Setup pydata version switcher
* note this means committing changes to the gh-pages branch rather that re initialising it each time

### Testing 

This has been tested on my personal repo, see https://github.com/samtygier-stfc/mantidimaging-tests/pull/2
This results in https://samtygier-stfc.github.io/mantidimaging-tests/index.html

### Acceptance Criteria 

Check https://samtygier-stfc.github.io/mantidimaging-tests/
Check that you can switch back and forth to 2.7.0 with the switcher in the top right
Check that you can switch to older versions (still on old theme)

### Documentation

Release notes

### CODE REVIEW

**Comment by samtygier-stfc** [comment]
The publish steps only run when merged to main or release branches. From release branches the changes are put in a sub folder.

On a PR we get
image

On main we get
image

**Comment by samtygier-stfc** [comment]
After this is merged, I repeat https://github.com/samtygier-stfc/mantidimaging-tests/pull/5 to update the 2.7 docs.

File: docs/_static/version_switcher.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
+[
+    {
+        "name": "v2.8.0 (devel)",
```
Comment (JackEAllen) [inline docs/_static/version_switcher.json:3]: NITPIC, I feel just calling this "v2.8.0 (dev)" may be better if just trying to shorten the word develop

**Review by JackEAllen — COMMENTED** [review]
Looks good, maybe just one typo to resolve. 
I'll write up an issue to print the version of MI selected from the dropdown in the dropdown to make the selected version a little more visible going forward.

**Review by JackEAllen — APPROVED** [review]

**Pull request merged**